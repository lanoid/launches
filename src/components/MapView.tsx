import * as React from 'react';
import * as L from 'leaflet';
import { Map, TileLayer, Marker } from 'react-leaflet';

interface Props {
    launchData?: any;
    position?: [number, number];
}

const MapView: React.FunctionComponent<Props> = (props) => {
    const bounds: [number, number][] = [];
    const icon = L.icon({
        iconUrl: require('../images/space-rocket-icon.png'),
        iconSize:     [20, 20],
        iconAnchor:   [20, 10]       
    })
    const Markers = props.launchData.map((launch: any, i: number) => {
        bounds.push([launch.location.pads[0].latitude, launch.location.pads[0].longitude]);
        return (
            <Marker key={i} position={[launch.location.pads[0].latitude, launch.location.pads[0].longitude]} icon={icon}></Marker>
        )
    });
    return (
        <Map center={props.position} zoom={13} bounds={bounds}>
            <TileLayer
                attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
                url='https://{s}.tile.osm.org/{z}/{x}/{y}.png'
            />
            {Markers}
        </Map>
    );
};

MapView.defaultProps = {
    position: [51.000, 0.000]
};

export default MapView;