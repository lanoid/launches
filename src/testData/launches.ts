const launches = [
    {
        name: 'Daft Punk',
        location: {
            name: 'My house',
            pads: [
                {
                    lat: 50,
                    long: 0
                }
            ]
        },
        windowstart: 'November 14, 2017 18:28:00 UTC',
        windowend: 'November 14, 2017 18:53:00 UTC'
    },
    {
        name: 'Daft Punk',
        location: {
            name: 'My house',
            pads: [
                {
                    lat: 51,
                    long: 0
                }
            ]
        },
        windowstart: 'November 14, 2017 18:28:00 UTC',
        windowend: 'November 14, 2017 18:53:00 UTC'
    },
    {
        name: 'Daft Punk',
        location: {
            name: 'Your house',
            pads: [
                {
                    lat: 52,
                    long: 0
                }
            ]
        },
        windowstart: 'November 14, 2017 18:28:00 UTC',
        windowend: 'November 14, 2017 18:53:00 UTC'
    }
];

export default launches;