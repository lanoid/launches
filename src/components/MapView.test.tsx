import * as React from 'react';
import { shallow } from 'enzyme';
import MapView from './MapView';
import * as launches from '../testData/launches';

describe('<MapView />', () => {
    const mapView = shallow(<MapView launchData={launches.default} />);
    
    it('matches the snapshot', () => {
        expect(mapView).toMatchSnapshot();
    });
});