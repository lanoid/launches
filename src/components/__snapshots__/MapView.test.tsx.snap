// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapView /> matches the snapshot 1`] = `
<Map
  bounds={
    Array [
      Array [
        undefined,
        undefined,
      ],
      Array [
        undefined,
        undefined,
      ],
      Array [
        undefined,
        undefined,
      ],
    ]
  }
  center={
    Array [
      51,
      0,
    ]
  }
  zoom={13}
>
  <ForwardRef(Leaflet(TileLayer))
    attribution="Â© <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors"
    url="https://{s}.tile.osm.org/{z}/{x}/{y}.png"
  />
  <ForwardRef(Leaflet(Marker))
    icon={
      NewClass {
        "_initHooksCalled": true,
        "options": Object {
          "iconAnchor": Array [
            20,
            10,
          ],
          "iconSize": Array [
            20,
            20,
          ],
          "iconUrl": "test-file-stub",
        },
      }
    }
    key="0"
    position={
      Array [
        undefined,
        undefined,
      ]
    }
  />
  <ForwardRef(Leaflet(Marker))
    icon={
      NewClass {
        "_initHooksCalled": true,
        "options": Object {
          "iconAnchor": Array [
            20,
            10,
          ],
          "iconSize": Array [
            20,
            20,
          ],
          "iconUrl": "test-file-stub",
        },
      }
    }
    key="1"
    position={
      Array [
        undefined,
        undefined,
      ]
    }
  />
  <ForwardRef(Leaflet(Marker))
    icon={
      NewClass {
        "_initHooksCalled": true,
        "options": Object {
          "iconAnchor": Array [
            20,
            10,
          ],
          "iconSize": Array [
            20,
            20,
          ],
          "iconUrl": "test-file-stub",
        },
      }
    }
    key="2"
    position={
      Array [
        undefined,
        undefined,
      ]
    }
  />
</Map>
`;
